test/build.test.js

```diff
@@ -0,0 +1,231 @@
+  const { build, stubs } = setup()
+  const data = clone(DEFAULT_ACTION_DATA)
+
+  data.inputs['build-command'] = 'npm install\n npm build'
+  data.inputs['app-name'] = APP_NAME
+  data.context.payload.pull_request.body =
+    '<!--\n' + '<release-meta>{"invalidjson}</release-meta>\n' + '-->'
+
+  try {
+    await build(data)
+  } catch (err) {
+    t.ok('Build expected to throw with invalid metadata')
+  }
+  sinon.assert.calledOnce(stubs.coreStub.setFailed)
+  sinon.assert.notCalled(stubs.execStub.exec)
+})
+
+tap.test('Should fail if running build command throws an error', async () => {
+  const { build, stubs } = setup()
+
+  const data = clone(DEFAULT_ACTION_DATA)
+  data.inputs['build-command'] = 'npm install\n npm build'
+
+  stubs.execStub.exec.throws()
+
+  await build(data)
+
+  sinon.assert.calledOnce(stubs.coreStub.setFailed)
+})
+
+tap.test(
+  'Should deal with multiple spaces in commands by removing them',
+  async () => {
+    const { build, stubs } = setup()
+
+    const data = clone(DEFAULT_ACTION_DATA)
+
+    data.inputs['build-command'] = 'npm  install\n npm build'
+
+    await build(data)
+
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'npm', ['-v'], {
+      cwd: '.',
+    })
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'node', ['-v'], {
+      cwd: '.',
+    })
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'npm', ['install'], {
+      cwd: '.',
+    })
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'npm', ['build'], {
+      cwd: '.',
+    })
+  }
+)
+
+tap.test(
+  'Should deal with multiple empty lines between commands, removing them',
+  async () => {
+    const { build, stubs } = setup()
+
+    const data = clone(DEFAULT_ACTION_DATA)
+
+    data.inputs['build-command'] = 'npm  install\n\n\n npm   build'
+
+    await build(data)
+
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'npm', ['-v'], {
+      cwd: '.',
+    })
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'node', ['-v'], {
+      cwd: '.',
+    })
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'npm', ['install'], {
+      cwd: '.',
+    })
+    sinon.assert.calledWithMatch(stubs.execStub.exec, 'npm', ['build'], {
+      cwd: '.',
+    })
+  }
+)
+
+tap.test(
+  'Should get inputs correctly when called from workflow_dispatch and ignore action metadata',
+  async () => {
+    const { build, stubs } = setup()
+
+    const data = clone(DEFAULT_ACTION_DATA)
+    data.github.event_name = 'workflow_dispatch'
+
+    data.inputs['monorepo-package'] = 'react-app'
+    data.inputs['monorepo-root'] = 'packages'
+    data.inputs['build-command'] = 'npm  install\n\nnpm   build'
+
+    data.context.payload.pull_request.body =
+      '<!--\n' +
+      '<release-meta>{"id":54503465,"version":"v5.1.3","npmTag":"latest", "monorepoPackage": "another-package", "monorepoRoot": "packages"}</release-meta>\n' +
+      '-->'
+
+    const cwd = `${data.inputs['monorepo-root']}/${data.inputs['monorepo-package']}`
```

Comment:
as far as I can see
- monorepo-root is only used here
- monorepo-package could contain a slash

...which kind of makes monorepo-root redundant in its current form? (other than my point above about supporting monorepos in a sub folder)